### ISSUE TITLE
IntelliSense incorrectly flags usage of foreach statement on type with multiple GetEnumerator extension methods

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#51337]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity2.visualstudio.com/t/IntelliSense-incorrectly-flags-usage-of-/1337486)._

---
[severity:It bothers me. A fix would be nice]
C# 9.0 enables extension `GetEnumerator` support for `foreach` loops:

[What's new in C# 9.0] (https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9)

If there is no instance `GetEnumerator` method and there is more than one reachable `GetEnumerator` extension method, the C# compiler correctly chooses the closest one. For instance:

```c#
using System.Collections;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using UnitTestProject1;

namespace UnitTestProject1
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            var c = new C();
            foreach (var i in c) ;
        }
    }

public class C { }

public static class Extensions
    {
        public static IEnumerator GetEnumerator(this C c) => new int[0]. GetEnumerator(); // The compiler chooses this extension method.
    }
}

public static class Extensions
{
    public static IEnumerator GetEnumerator(this C c) => new int[0]. GetEnumerator();
}
```

The previous example compiles and runs just fine. However, IntelliSense incorrectly flags the usage of variable `c` in the `foreach` loop, stating:

`Type 'UnitTestProject.C' cannot be used in 'foreach' statement because it neither implements 'IEnumerable' or 'IEnumerable<T>', nor has suitable 'GetEnumerator' method which return type has 'Current' property and 'MoveNext' method`

Removing either of the extension methods causes the flag to go away.

---
#### Feedback Bot on 2/11/2021, 09:35 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by genlu** [comment]
Looks like a compiler issue.

**Comment by jaredpar** [comment]
@genlu why do you think it's a compiler issue, the code builds so compiler has correct behavior here. Assumption would be that the IDE hosting of the compiler is not providing all the right info. 

If we can't repro though then we should resolve. Can reactivate if we see it again later.

**Comment by muelljon** [comment]
@genlu @jaredpar 

My apologies, you are correct. This is not a Visual Studio/IntelliSense issue. I just figured out this is a bug in an extension I have installed (ReSharper) that made the errors look exactly like IntelliSense errors. Please disregard.
